src/modules/facility-fixed-fee/facility-fixed-fee.service.test.ts

```diff
@@ -0,0 +1,58 @@
+import { PROPERTIES } from '@ukef/constants';
+import { AcbsAuthenticationService } from '@ukef/modules/acbs-authentication/acbs-authentication.service';
+import { DateStringTransformations } from '@ukef/modules/date/date-string.transformations';
+import { getMockAcbsAuthenticationService } from '@ukef-test/support/abcs-authentication.service.mock';
+import { GetFacilityFixedFeeGenerator } from '@ukef-test/support/generator/get-facility-fixed-fee-generator';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { when } from 'jest-when';
+
+import { AcbsFacilityFixedFeeService } from '../acbs/acbs-facility-fixed-fee.service';
+import { FacilityFixedFeeService } from './facility-fixed-fee.service';
+
+describe('FacilityFixedFeeService', () => {
+  const { portfolioIdentifier } = PROPERTIES.GLOBAL;
+  const valueGenerator = new RandomValueGenerator();
+  const idToken = valueGenerator.string();
+  const facilityIdentifier = valueGenerator.facilityId();
+
+  const { facilityFixedFees: expectedFacilityFixedFees, facilityFixedFeesInAcbs } = new GetFacilityFixedFeeGenerator(
+    valueGenerator,
+    new DateStringTransformations(),
+  ).generate({ numberToGenerate: 2, facilityIdentifier, portfolioIdentifier });
+
+  let acbsAuthenticationService: AcbsAuthenticationService;
+  let service: FacilityFixedFeeService;
+
+  let getFacilityFixedFeesAcbsService: jest.Mock;
+
+  beforeEach(() => {
+    const mockAcbsAuthenticationService = getMockAcbsAuthenticationService();
+    acbsAuthenticationService = mockAcbsAuthenticationService.service;
+    const acbsAuthenticationServiceGetIdToken = mockAcbsAuthenticationService.getIdToken;
+    when(acbsAuthenticationServiceGetIdToken).calledWith().mockResolvedValueOnce(idToken);
+
+    const acbsService = new AcbsFacilityFixedFeeService(null, null);
+    getFacilityFixedFeesAcbsService = jest.fn();
+    acbsService.getFixedFeesForFacility = getFacilityFixedFeesAcbsService;
+
+    service = new FacilityFixedFeeService(acbsAuthenticationService, acbsService, new DateStringTransformations());
+  });
+
+  describe('getFixedFeesForFacility', () => {
+    it('returns a transformation of the guarantees from ACBS', async () => {
```

Comment:
Can we add following assertions:

1. 404
2. Invalid facility ID i.e. `ABCD` or `!£!"£`.

Reply:
1. See above re 404 (no longer needed)
2. This cannot be tested at a service level as it is a validation that happens at controller level (`ValidatedFacilityIdentifierApiProperty` on `get-facility-fixed-fee-params.dto.ts`)